lib/matplotlib/axes/_axes.py

```diff
@@ -6027,8 +6027,11 @@ def _pcolorargs(self, funcname, *args, shading='auto', **kwargs):
                 # grid is specified at the center, so define corners
                 # at the midpoints between the grid centers and then use the
                 # flat algorithm.
-                def _interp_grid(X):
-                    # helper for below
+                def _interp_grid(X, require_monotonicity=False):
+                    # helper for below. To ensure the cell edges are calculated
+                    # correctly, when expanding columns, the monotonicity of
+                    # X coords needs to be checked. When expanding rows, the
+                    # monotonicity of Y coords needs to be checked.
```

Comment:
I think this fix is correct, but you never use require_monotinicity

Reply:
Thank you. I've already added "require_monotonicity" to the judgment conditions. Thanks for your suggestion. I'll try to deploy the test environment locally.

Reply:
Note you don't really need to provision yourself with a whole test environment if your test is simple enough.  In this case I think you are just testing for the presence of a warning, so adding a small test in `test_axes.py` should be fine, and could probably be checked using the automated CI testing on GitHub.

Reply:
Thank you for your suggestion. I will try to learn how to add a test.